experimental/Schemes/Resolution_structure.jl

```diff
@@ -0,0 +1,754 @@
+  return f
+end
+
+########################################################################
+# Refinements to find local systems of parameters
+########################################################################
+
+function find_refinement_with_local_system_of_params(W::AbsAffineScheme{<:Field, <:MPolyRing}; check::Bool=true)
+  U = PrincipalOpenSubset(W, one(OO(W)))
+  res_cov = Covering([U])
+  R = ambient_coordinate_ring(W)
+  minor_dict = IdDict{AbsAffineScheme, Tuple{Vector{Int}, Vector{Int}, elem_type(R)}}()
+  minor_dict[U] = (Int[], Int[], one(R))
+  return res_cov, minor_dict
+end
+
+function find_refinement_with_local_system_of_params(W::AbsAffineScheme; check::Bool=true)
+  @check is_smooth(W) "scheme must be smooth"
+  @check is_equidimensional(W) "scheme must be equidimensional"
+  mod_gens = lifted_numerator.(gens(modulus(OO(W))))::Vector{<:MPolyRingElem}
+  # We run into difficulties for the zero ideal as a modulus.
+  # To get the matrix of minors of jac(I) we use `induced_map_on_exterior_power` below.
+  # It is mathematical convention that ⋀⁰R⁰ = R¹. But that's unfortunately not 
+  # coherent with the generic indexing we use in the implementation below. 
+  # Should this assertion lead to problems, one can still replace throwing an error 
+  # by inserting the shortcut from the method above and returning that. 
+  @assert !isempty(mod_gens) "method not implemented for empty modulus; try to create the scheme without modulus instead"
+  R = ambient_coordinate_ring(W)
+  M = jacobian_matrix(R, mod_gens)
+
+  n = nrows(M) # the number of variables in the ambient_ring
+  r = ncols(M) # the number of generators
+
+  Rn = FreeMod(R, n)
+  Rr = FreeMod(R, r)
+  phi = hom(Rn, Rr, M)
+  codim = n - dim(W)
+  phi_cod = induced_map_on_exterior_power(phi, codim)
+  M_ext = matrix(phi_cod)
+  n_cod = nrows(M_ext)
+  r_cod = ncols(M_ext)
+
+  all_entries = Vector{Int}[[i, j] for i in 1:n_cod for j in 1:r_cod]
+  M_ext_vec = elem_type(R)[M[i, j] for i in 1:n_cod for j in 1:r_cod]
+  min_id = ideal(OO(W), M_ext_vec)
+  lambda_vec = coordinates(one(OO(W)), min_id)
+  lambda = elem_type(OO(W))[lambda_vec[(i-1)*r_cod + j] for i in 1:n_cod, j in 1:r_cod]
+  
+  nonzero_indices_linear = [k for k in 1:length(lambda_vec) if !is_zero(lambda_vec[k])]
+  non_zero_indices = [[i, j] for i in 1:n_cod, j in 1:r_cod if !is_zero(lambda_vec[(i-1)*r_cod + j])]
+
+  ref_patches = AbsAffineScheme[]
+  minor_dict = IdDict{AbsAffineScheme, Tuple{Vector{Int}, Vector{Int}, elem_type(R)}}()
+  for (i, j) in non_zero_indices
+    h_ij = M_ext[i, j]
+    U_ij = PrincipalOpenSubset(W, OO(W)(h_ij))
+    I = ordered_multi_index(i, codim, n)
+    J = ordered_multi_index(j, codim, r)
+    push!(ref_patches, U_ij)
+    minor_dict[U_ij] = (indices(I), indices(J), M_ext[i, j])
+  end
+  res_cov = Covering(ref_patches)
+  inherit_glueings!(res_cov, Covering(W))
+  return res_cov, minor_dict
+ #=
+  all_entries = Vector{Int}[[i, j] for i in 1:n for j in 1:r]
```

Comment:
These lines can probably go now?

Reply:
Feel free to throw them away. You were the one who wanted to keep them in the first place.

Reply:
Yes, I remember. But I wasn't sure whether this internal helper function had already grown to everyone's full satisfaction, or whether we still wanted to sample from that snippet. Thanks for the cleanup.